sync/awsdatabricks.py

```diff
@@ -350,6 +351,61 @@ def _get_aws_cluster_info_from_s3(bucket: str, file_key: str, cluster_id):
         logger.warning(f"Failed to retrieve cluster info from S3 with key, '{file_key}': {err}")
 
 
+def save_cluster_report(
+    cluster_id: str,
+    instance_timelines: list[dict],
+    cluster_log_destination: tuple[str, ...] | None = None,
+    cluster_report_destination_override: dict[str, str] | None = None,
+) -> bool:
+    cluster = get_default_client().get_cluster(cluster_id)
+    spark_context_id = cluster.get("spark_context_id")
+
+    if not spark_context_id:
```

Comment:
@brandon-kaplan just to double check, spark_context_id only exists when the cluster is running right?

I may want to not check for eventbridge cause we could process events after the cluster is done running.

Do you see any caveats with saving the cluster report after the cluster is done running? Below `_get_ec2_instances` fetches the current running instances, I'll take a look on Monday.

Reply:
it gets a value when the cluster starts running but it doesn't disappear after its completed, ill ping you directly since I'm replying so late